portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/FormEvents.testcase

```diff
@@ -65,6 +65,175 @@ definition {
 		ACUtils.tearDownAC();
 	}
 
+	@description = "Bug: LRAC-7832 | Automation ID: LRAC-9867 | Test Summary: Check fieldBlurred and fieldFocused is triggered when interacting with form fields (Text, Date, Date and Time, Image, Rich Text, Upload and Color[by color picker and color field])"
+	@priority = "4"
+	test CheckFieldBlurredAndFieldFocusedWhenInteractingWithFields {
+		property proxy.server.enabled = "true";
+
+		var fieldNameList = "Date,Image,Rich Text,Upload,Color,Date and Time";
+
+		task ("Edit the form to add more fields") {
+			FormsAdmin.openFormsAdmin(siteURLKey = "site-name");
+
+			FormsAdminNavigator.gotoEditFormByName(formName = "Created Form Name");
+
+			DataEngine.addFields(fieldNameList = "${fieldNameList}");
+
+			Form.save();
+		}
+
+		task ("Navigate to view the form page") {
+			ACUtils.navigateToSitePage(
+				actionType = "View Form",
+				pageName = "AC Page",
+				siteName = "Site Name");
+		}
+
+		task ("This loop will interact with the input field of all fields") {
+			for (var fieldName : list "Text,${fieldNameList}") {
+				task ("Start Har recording") {
+					ProxyUtil.startHarRecording("fieldFocused");
+				}
+
+				task ("Click to focus on form fields different than rich text") {
+					if ("${fieldName}" == "Date and Time") {
+						Click(
+							key_fieldName = "DateAndTime",
+							locator1 = "FormFields#GENERIC_INPUT_FIELD");
+					}
+
+					if (("${fieldName}" != "Rich Text") && ("${fieldName}" != "Date and Time")) {
+						Click(
+							key_fieldName = "${fieldName}",
+							locator1 = "FormFields#GENERIC_INPUT_FIELD");
+					}
+				}
+
+				task ("Click to focus on rich text writing field") {
+					if ("${fieldName}" == "Rich Text") {
+						FormFields.addRichTextContent(
+							content = "",
+							fieldLabel = "Rich Text");
+					}
+				}
+
```

Comment:
Hey @ViniLopes20 can you tell me if I'm understanding this block of code right?
So we have an `if`just for `Date and Time` :ok: 
We have an `if`just when the fields are different than `Rich Text` and `Date and Time`, so this leaves us to check for: `Date,Image,Upload,Color` :ok: 
and the last `if` checks for `Rich Text` :ok: 
I found it a little bit confusing, do you think using `else if` statements (like [this example here](https://github.com/liferay/liferay-portal/blob/master/portal-web/test/functional/com/liferay/portalweb/tests/enduser/documentmanagement/documentsadministration/DMAdmin.testcase#L241#L255)) would be better? I know the code will get a little bit bigger, but I want to know your opinion on it.

Reply:
Hey @manuferreira, I remember Alex had told me that he thinks using **else if** or **else** is a little bad for code legibility. He gave me this [video here](https://www.youtube.com/watch?v=EumXak7TyQ0). I think our situation is different from the case of the video that will use a return to not have to do all the conditions. So I think we could use **else if** here to save some of the Poshi resources. But I think in relation to which one is easier to understand, I think using the **if** makes it a little easier to read the test logic.

Reply:
We could also use the **else** in this case, but I think that the use of the **else** will complicate the reading of the code that a person could have, the use of the **else** does not make it so clear which cases will be included in it, so I think that we shouldn't use in this situation to be clear what the fields are.